File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cs
public class Solution {
    public int FindMin(int[] nums) {
     int min = 0;
     int l =0;
     int r = nums.Length -1;
     while(l <=r){
        if(nums[l] <= nums[r]){
            min = nums[l];
            break;
        }
        int mid = l+ (r-l) /2;
        if(nums[mid] >= nums[l]){
            l = mid+1;
        }
        else{
            r = mid;
        }
     }
     return min;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-1.ts
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights: number[]): number {
        let l=0;
        let r  = heights.length -1;
        let maxWater = 0;
        while( l < r){
          const waterCon = (r-l) * Math.min(heights[l],heights[r]);
          if(heights[l] > heights[r]) r--;
          else if(heights[l] < heights[r]) l++;
          else{
            r--;
            l++;
          }
           maxWater = Math.max(maxWater,waterCon);
        }
       return maxWater;
    }
}
